msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/IMDSManagedIdentity.java

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
```

Comment:
The flat packaging in MSAL 4J is not great. Makes the code hard to navigate. Can you create a "folder" at least for MI?

Reply:
I am not sure if this can be done. The issue will be having to reference those classes out of that package since the scope will be package private and not public.

Reply:
This is something that's been talked about before, I'm not sure why it was originally designed this way but it certainly can be confusing. 

For now I think it's best to keep the MI implementation as similar to the rest of the library as possible, and focus on improving the entire library structure in another PR.

Reply:
Agree this is something that will not be a breaking change and can be achieved in another PR.